aoc2022/src/main/scala/day05.scala

```diff
@@ -0,0 +1,90 @@
+import scala.io.*
+import math.*
+import scala.annotation.tailrec
+
+/**
+ * PART 01:
+ *
+ * The tricky part of this day was to correctly parse in the input file. Especially the stacks are hard to parse. I 
+ * ended up replacing the spaces between containers with empty containers, like this "[]". Then the string was easy
+ * to parse. To link each container stack to its correct ID, I transposed the List of containers, like so: 
+ * 
+ * List(List(, D, ), List(N, C, ), List(Z, M, P), List(1, 2, 3))
+ * 
+ * to:
+ * 
+ * List(List(, N, Z, 1), List(D, C, M, 2), List(, , P, 3)) 
+ * 
+ * Then I converted it in a map like this: Map(1 -> List(N, Z), 2 -> List(D, C, M), 3 -> List(P))
+ * 
+ * After parsing the input stack, it was trivial to create the logic of changing a stack state to the next stated 
+ * based on a move. Result is obtained by foldLeft on every move with the stack as fold result. 
+ *
+ * PART 02:
+ *
+ * Changed the exectuteMove function to also work with the non-reversed order for part02. 
+ *
+ */
+
+
+object day05 extends App:
+
+  private val day: String =
+    this.getClass.getName.drop(3).init
+
+  private val start1: Long =
+    System.currentTimeMillis
+
+  case class Move(qty: Int, src: Int, tgt: Int)
+  
+  private val (stack, moves): (Map[Int, List[Char]], List[Move]) =
+
+    def parseStacks(s: String): List[String] =
+      if s.contains('[') then
+        val stackRow: String = s.replace("    ", " []")
+        stackRow.split(" ").map(_.replace("[", "").replace("]", "")).toList
```

Comment:
The question is how generic your solutions needs to be. A less generic option is to use `.grouped(4)` which probably results code that is more readable, but less flexible (only single letter stacks are supported).

Reply:
Nice, I implemented .grouped and indeed it's more readable. I used the entire container, including square brackets, as key for the Map so that it will always run independent on the amount of letters per container.